### PR TITLE
Adding ability to send metrics to influxdb

### PR DESCRIPTION
**Pull request by joejstuart**

### CODE REVIEW

File: vars/ciMetrics.groovy [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+} finally {
+    ciMetrics.writeToInflux()
+}
+ */
+class ciMetrics {
+
+    // A map to store the data sent to influx
+    def customDataMap = ["ci_pipeline": [:]]
+    // This will prefix the data sent to influx. Usually set to the job name.
+    def prefix = "ci_pipeline"
```
Comment (robnester-rh) [inline vars/ciMetrics.groovy]: I'd perhaps set this to default to the job name, and allow overiding?

File: vars/ciMetrics.groovy [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    // The influx target configured in jenkins
+    def influxTarget = "localInflux"
+    // The influx database that will be written to
+    def measurement = "ci_pipeline"
```
Comment (robnester-rh) [inline vars/ciMetrics.groovy]: Could this be configured such that it is based on the job name as well? Or would it make more sense to leave it as a non-variable database name? I'm not sure what your influx database provisioning code looks like.

File: vars/ciMetrics.groovy (hunk 1 above)
Comment (joejstuart) [reply vars/ciMetrics.groovy]: Good suggestion. I will update this.

Comment (joejstuart) [reply vars/ciMetrics.groovy]: I'm going to leave the defaults as is for now. Maybe just an update to the docs. Since this will prefix the job name.

**Pull request merged**